### ISSUE TITLE
A function's static variables cannot be accessed through the function name anymore

### ISSUE THREAD

**Issue by YYBartT** [YoYoGames/GameMaker-Bugs#4403]
### Description

A function's static variables should be accessible through the function name once the function's been called at least once. This seems to be broken in the current beta (Beta IDE v2024.200.0.461 Beta Runtime v2024.200.0.477).

As an example, the following code on the [Static Struct](https://manual.gamemaker.io/monthly/en/GameMaker_Language/GML_Overview/Structs/Static_Structs.htm) page now throws an error: 
```gml
function counter() {
    static count = 0;
    return count ++;
}

repeat (10) counter()

// Get static struct of counter()
var _static_counter = static_get(counter);

// Both of these read the same variable
show_debug_message(counter.count); // 10
show_debug_message(_static_counter.count); // 10
```

The line: 
```gml
show_debug_message(counter.count); // 10
```
shows an error `Variable <unknown_object>.count(100003, -2147483648) not set before reading it.`, instead of displaying the static variable's value.

Access to the variable through the static struct still works fine.

### Expected Change

Access to a function's static variables through the function name works again

### Steps To Reproduce

1. Create a new project in GM
2. Copy-paste the example code in a script asset
3. Run the game
4. The error message is shown

### How reliably can you recreate this issue using your steps above?

Always

### Which version of GameMaker are you reporting this issue for?

2024.200 (Betas)

### Which operating system(s) are you seeing the problem on?

Windows 10

### Are you running GameMaker from inside your Steam library?

No

### Contact Us Package Attached?

- [X] I have attached my Contact Us Package

### Sample Project Added?

- [ ] I have included a small sample project

**Comment by rwkay** [comment]
Fixed in 2024.2

**Comment by cameron-home** [comment]
Verified fixed in IDE v2024.200.0.469  Runtime v2024.200.0.485
